android/src/main/java/com/adyenreactnativesdk/util/ReactNativeError.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2022 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ */
+package com.adyenreactnativesdk.util
+
+import android.util.Log
+import com.adyenreactnativesdk.component.BaseModuleException
+import com.adyenreactnativesdk.component.KnownException
+import com.adyenreactnativesdk.component.dropin.DropInException
+import com.facebook.react.bridge.ReadableMap
+import com.facebook.react.bridge.WritableMap
+import com.facebook.react.bridge.WritableNativeMap
+import java.lang.Exception
+
+object ReactNativeError {
+
+    private const val MESSAGE_KEY = "message"
+    private const val REASON_KEY = "reason"
+    private const val DISCRIPTION_KEY = "discription"
+    private const val RECOVERY_KEY = "recovery"
+    private const val ERROR_CODE = "errorCode"
+
+    fun mapError(message: String?): ReadableMap {
```

Comment:
(optional) you can make this function look more kotlinish and short by doing it like this: 

```
    fun mapError(message: String?): ReadableMap {
        return WritableNativeMap().apply {
            putString(MESSAGE_KEY, message)
        }
    }
```

Reply:
you can also do something similar for the function below

Reply:
👍

Reply:
Is this Kotlin enough? 😁

```kotlin
fun mapError(error: Exception): ReadableMap {
        return WritableNativeMap().apply {
            putString(MESSAGE_KEY, error.localizedMessage)
            error.cause?.let {
                putString(REASON_KEY, it.localizedMessage)
            }
            error.stackTrace
                ?.takeIf { it.isNotEmpty() }
                ?.let { putString(DISCRIPTION_KEY, it.toString()) }

            (error as? KnownException)?.let {
                putString(ERROR_CODE, it.code)
            }
        }
    }
```

Reply:
yes but i'd prefer keeping checks as they are using `if` statements for readability in this one. you can keep the `.apply` part though :)

Reply:
Fixed in [a36788f](https://github.com/Adyen/adyen-react-native/pull/52/commits/a36788f155b58a5d4450d69788b3a20930326f71)